src/hooks/xcm/parachainApi/MoonbeamApi.ts

```diff
@@ -0,0 +1,201 @@
+import { LOCAL_STORAGE } from './../../../config/localStorage';
+import { EthereumProvider } from './../../types/CustomSignature';
+import { u8aToHex } from '@polkadot/util';
+import { decodeAddress } from '@polkadot/util-crypto';
+import BN from 'bn.js';
+import { supportEvmWallets, SupportWallet } from 'src/config/wallets';
+import { getTokenBal, setupNetwork } from 'src/config/web3';
+import moonbeamXcmAbi from 'src/config/web3/abi/moonbeam-xcm-abi.json';
+import { wait } from 'src/hooks/helper/common';
+import { Asset } from 'src/v2/models';
+import Web3 from 'web3';
+import { TransactionConfig } from 'web3-eth';
+import { AbiItem } from 'web3-utils';
+import { RelaychainApi } from '../SubstrateApi';
+import { isValidEvmAddress } from './../../../config/web3/utils/convert';
+import { getEvmProvider } from './../../helper/wallet';
+import { ethers } from 'ethers';
+
+type chainName = 'Moonriver' | 'Moonbeam';
+
+// Ref: https://docs.moonbeam.network/builders/build/canonical-contracts/precompiles/erc20/
+const NATIVE_TOKEN_ADDRESS = '0x0000000000000000000000000000000000000802';
+const PRE_COMPILED_ADDRESS = '0x0000000000000000000000000000000000000804';
+
+export class MoonbeamApi extends RelaychainApi {
```

Comment:
Why is Moonbeam (a parachain) extending from the relaychain API?

Reply:
I've fixed in this commit. https://github.com/AstarNetwork/astar-apps/pull/471/commits/338cd751a0aa511df01d0c6f9d63c605fea82244